packages/ra-core/src/auth/useLogoutIfAccessDenied.spec.tsx

```diff
@@ -258,4 +258,36 @@ describe('useLogoutIfAccessDenied', () => {
             expect(screen.queryByText('unauthorized')).not.toBeNull();
         });
     });
+
+    it('should stay on same page if error have no redirectTo', async () => {
+        render(
+            <TestMemoryRouter>
+                <AuthContext.Provider
+                    value={{
+                        ...authProvider,
+                        checkError: () => {
+                            return Promise.reject({
+                                logoutUser: false,
+                                message: 'Access denied',
+                            });
+                        },
+                    }}
+                >
+                    <Routes>
+                        <Route path="/" element={<TestComponent />} />
+                        <Route path="/login" element={<div>Login page</div>} />
+                    </Routes>
+                </AuthContext.Provider>
+            </TestMemoryRouter>
+        );
+
+        await waitFor(() => {
+            expect(authProvider.logout).toHaveBeenCalledTimes(0);
+            expect(notify).toHaveBeenCalledWith('Access denied', {
+                type: 'error',
+            });
+            expect(notify).toHaveBeenCalledTimes(1);
+            expect(screen.queryByText('Login page')).toBeNull();
+        });
```

Comment:
This test is probably not very robust, because you don't wait before checking that `'Login page'` is null. I suspect this tests would pass even without the fix. Can you wait for some time to let the redirection happen (if any) before checking that we are not redirected to the login page?

Reply:
Thanks, but this is not what I meant. `'Login page'` will be `null` immediately because you are not on the login page at first. Giving more time to react testing lib to test that is not gonna change the fact that the test will be valid immediately.

What you need to do instead is wait some time (e.g. 50 ms) **_before_** trying to check if  `'Login page'`  is present on the screen.

You can try this the other way around: try to write a test that fails without the fix, then re-apply the fix and confirm the test passes.